emcc.py

```diff
@@ -2394,7 +2386,7 @@ def generate_html(target, options, js_target, target_basename,
 %s
           };
           emterpretXHR.send(null);
-''' % (shared.Settings.EMTERPRETIFY_FILE, script.inline)
+''' % (shared.JS.get_subresource_location(shared.Settings.EMTERPRETIFY_FILE), script.inline)
```

Comment:
I took care of this as suggested by @curiousdannii. However, while doing this, I noticed that this part of emcc.py injects a number of one-off implementations of basically the same logic from `Module.read*`. Is there a reason that I shouldn't change them to use `Module.read*`instead of reimplementing the data URI parsing in each one?